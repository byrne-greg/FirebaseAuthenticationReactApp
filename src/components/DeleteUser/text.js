export default {
  deleteYourAccount: "Delete your account",
  deleteYourAccountQuestion: "Are you sure you want to delete your account?",
  deleteAccount: "Delete account",
  accountDeleted: "Account deleted",
  accountIsDeleted: "You have successfully deleted your account"
};
