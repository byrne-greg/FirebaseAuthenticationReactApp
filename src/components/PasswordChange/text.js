export default {
  changeYourPassword: "Change your password",
  changePassword: "Change Password",
  newPassword: "New Password",
  confirmNewPassword: "Confirm New Password"
};
