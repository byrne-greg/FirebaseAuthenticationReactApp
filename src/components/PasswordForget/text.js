export default {
  emailAddress: "Email Address",
  resetMyPassword: "Reset My Password",
  passwordReset: "Password Reset",
  forgotYourPassword: "Forget your password?",
  forgotPassword: "Forgot Password?"
};
