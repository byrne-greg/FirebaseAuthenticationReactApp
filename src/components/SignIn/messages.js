export default {
  signIn: "Sign In",
  emailAddressLabel: "Email Address",
  email: "email",
  passwordLabel: "Password",
  password: "password"
};
