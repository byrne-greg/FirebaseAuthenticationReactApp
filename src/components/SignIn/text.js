export default {
  signIn: "Sign In",
  emailAddress: "Email Address",
  password: "Password"
};
