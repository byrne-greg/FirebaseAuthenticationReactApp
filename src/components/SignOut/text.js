export default {
  successfulSignOut: "You have successfully signed out",
  signOut: "Sign Out"
};
