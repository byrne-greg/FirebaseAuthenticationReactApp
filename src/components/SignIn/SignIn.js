import React, { useState, useContext } from "react";
import { withRouter } from "react-router-dom";

import { SignUpLink } from "../SignUp";
import { PasswordForgetLink } from "../PasswordForget";
import { FirebaseContext } from "../Firebase";
import * as ROUTES from "../../constants/routes";
import text from "./text";

const SignIn = () => (
  <div>
    <h1>{text.signIn}</h1>
    <SignInForm />
    <PasswordForgetLink />
    <SignUpLink />
  </div>
);

const ErrorAlert = ({ errorText }) => (
  <div id="submit-error">
    <p>{errorText}</p>
  </div>
);

const SubmitButton = ({ isInvalid }) => (
  <button id="submit-button" disabled={isInvalid} type="submit">
    {text.signIn}
  </button>
);

const EmailInput = ({ currentValue, onChange }) => (
  <>
    <label id="email-label">{text.emailAddress}</label>
    <input
      aria-labelledby="email-label"
      id="email-input"
      name="email"
      value={currentValue}
      onChange={onChange}
      type="email"
      placeholder={text.emailAddress}
    />
  </>
);

const PasswordInput = ({ currentValue, onChange }) => (
  <>
    <label id="password-label">{text.password}</label>
    <input
      aria-labelledby="password-label"
      id="password-input"
      name="password"
      value={currentValue}
      onChange={onChange}
      type="password"
      placeholder={text.password}
    />
  </>
);

const SignInFormComponent = ({ history }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState(null);
  const firebase = useContext(FirebaseContext);

  const isInvalid = !password || !email;

  const onSubmit = async event => {
    event.preventDefault();
    try {
      await firebase.login(email, password);
      setEmail("");
      setPassword("");
      history.push(ROUTES.HOME);
    } catch (error) {
      setError({ ...error });
    }
  };

  const handleChange = ({ event, setState }) => {
    setState(event.target.value);
    setError(null);
  };
  const onEmailChange = event => {
    handleChange({ event, setState: setEmail });
  };
  const onPasswordChange = event => {
    handleChange({ event, setState: setPassword });
  };

  return (
    <form onSubmit={onSubmit}>
      <EmailInput value={email} onChange={onEmailChange} />
      <PasswordInput value={password} onChange={onPasswordChange} />
      <SubmitButton isInvalid={isInvalid} />
      {error && <ErrorAlert errorText={error.message} />}
    </form>
  );
};
const SignInForm = withRouter(SignInFormComponent);

export default SignIn;
export { SignInForm };
