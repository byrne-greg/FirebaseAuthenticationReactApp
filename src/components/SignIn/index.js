import SignIn, { SignInForm } from "./SignIn";

export default SignIn;
export { SignInForm };
