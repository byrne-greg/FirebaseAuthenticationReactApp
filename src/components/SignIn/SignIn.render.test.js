import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { withTestRouter, withMockFirebase } from "../../TestUtil";
import SignIn, { SignInForm } from ".";
import text from "./text";

afterEach(cleanup);

describe("SignIn render tests", () => {
  it("page renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<SignIn />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  xit("form renders with error", () => {});
});
