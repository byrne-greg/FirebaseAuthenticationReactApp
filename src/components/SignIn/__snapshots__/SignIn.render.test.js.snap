// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn render tests page renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        Sign In
      </h1>
      <form>
        <label
          id="email-label"
        >
          Email Address
        </label>
        <input
          aria-labelledby="email-label"
          id="email-input"
          name="email"
          placeholder="Email Address"
          type="email"
          value=""
        />
        <label
          id="password-label"
        >
          Password
        </label>
        <input
          aria-labelledby="password-label"
          id="password-input"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          disabled=""
          type="submit"
        >
          Sign In
        </button>
      </form>
      <p>
        <a
          href="/pw-forget"
        >
          Forgot Password?
        </a>
      </p>
      <p>
        Don't have an account? 
        <a
          href="/signup"
        >
          Sign Up
        </a>
      </p>
    </div>
  </div>
</body>
`;
